### DISCUSSION TITLE
Is there a way for Seq to log the Incoming IP Address

### DISCUSSION THREAD

**Discussion by giblets** [datalust/seq-tickets#1382] [Ideas]
Is there a way that Seq can grab what it detects as the incoming IP Address and attach it to the message like it does with timestamps?  

It would be useful if we could add that as an option on the APIKey, to collect the originating IP Address of the message like we use server timestamps.
![image](https://user-images.githubusercontent.com/590086/142483124-17af2fff-5cf7-415b-9388-944ca7c7d649.png)

Would grabbing incoming the IP Address greatly negatively impact the performance?

**Comment by nblumhardt** [comment]
Hi Glenn, thanks for the note. We've considered this in the past; it's not so much a performance problem as added complexity in the ingestion path, which we keep a high bar for increasing. Are you trying to track down a rogue log source?

**Comment by giblets** [comment]
Yes we were considering it for that.  Also because it would nice for
geo-mapping where reporting is coming from on something like google maps or
open street maps.

**Reply by nblumhardt** [reply]
:+1: thanks for your reply. RE tracking down log sources, allocating fine-grained API keys works well for this. The geo-mapping possibilities are interesting, we'll give this some more consideration - thanks!

**Reply by ebioux-btig** [reply]
Hi Nick - have you given this more consideration? I can also see a fair amount of value!
Thanks!!

**Reply by nblumhardt** [reply]
Hi Emmanuel; is your scenario also driven by the need to track rogue log sources? Or more from the mapping angle? Thanks for the feedback!
